src/model/wallet.ts

```diff
@@ -388,21 +414,33 @@ export const signMessage = async (
   result?: string;
   error?: any;
 }> => {
+  let isHardwareWallet = false;
   try {
     logger.info('wallet: signing message', { message });
     const wallet =
       existingWallet || (await loadWallet(undefined, true, provider));
+    if (!(wallet instanceof Wallet)) {
+      isHardwareWallet = true;
+    }
     try {
       if (!wallet) return null;
       const result = await wallet.signMessage(arrayify(message));
       return { result };
     } catch (error) {
-      Alert.alert(lang.t('wallet.transaction.alert.failed_sign_message'));
+      if (isHardwareWallet) {
+        setHardwareTXError(true);
+      } else {
+        Alert.alert(lang.t('wallet.transaction.alert.failed_sign_message'));
+      }
       logger.error(new RainbowError('Failed to sign message'), { error });
       return { error };
     }
   } catch (error) {
-    Alert.alert(lang.t('wallet.transaction.alert.authentication'));
+    if (isHardwareWallet) {
+      setHardwareTXError(true);
+    } else {
+      Alert.alert(lang.t('wallet.transaction.alert.authentication'));
+    }
```

Comment:
I feel like this if could probably be wrapped in some function to not repeat the same if/else statement over and over. With a descriptive name it will still be very obvious and readable

Reply:
i had similar thoughts about the check `isHardwareWallet` logic that's repeated